src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Create.cs

```diff
@@ -381,5 +394,167 @@ private static void Free(bool ignoreCase, HashSet<string> identifiers, HashSet<s
             SharedPools.StringHashSet.ClearAndFree(identifiers);
             SharedPools.StringHashSet.ClearAndFree(escapedIdentifiers);
         }
+
+        private static void ProcessStringLiteralForGlobalAttributeInfo(
+            SyntaxToken stringLiteral,
+            ISyntaxFacts syntaxFacts,
+            PooledDictionary<string, ArrayBuilder<int>> globalAttributeInfoBuilder)
+        {
+            var text = stringLiteral.ValueText;
+            if (string.IsNullOrEmpty(text) ||
+                !IsTargetOfGlobalAttribute(stringLiteral, syntaxFacts))
+            {
+                return;
+            }
+
+            // Target string must contain a valid symbol DocumentationCommentId,
+            // with an optional '~' prefix.
+            // First walk past this optional prefix.
+            var docCommentId = text[0] == '~' ? text.Substring(1) : text;
```

Comment:
do we have to do this allocation now?  can we not just have the ~ in the string we store and handle that later when doing a match?

Reply:
no longer relevant